compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.desktop.kt

```diff
@@ -327,10 +327,10 @@ internal class ComposeContainer(
         onKeyEvent: (KeyEvent) -> Boolean = { false },
     ) {
         mediator.setKeyEventListeners(
-            onPreviewKeyEvent = {
-                onPreviewKeyEvent(it) || backGestureDispatcher.onKeyEvent(it)
-            },
-            onKeyEvent = onKeyEvent
+            onPreviewKeyEvent = onPreviewKeyEvent,
+            onKeyEvent = {
+                backGestureDispatcher.onKeyEvent(it) || onKeyEvent(it)
```

Comment:
Shouldn't we use only unconsumed by the app events?
```suggestion
                onKeyEvent(it) || backGestureDispatcher.onKeyEvent(it)
```

Reply:
<img width="500" alt="image" src="https://github.com/user-attachments/assets/d7241829-aa64-4ba7-ad6a-ad27b5b41d3b" />

As I understand, no.
The idea is to intercept a key event and if it is handled by BackHandler then not provide it further

Reply:
Currently it is basically the same as inside `onPreviewKeyEvent` after Compose. Like before this PR.

Reply:
The handler here is attached to the window/dialog. I'm not sure which behavior is better, or even any reason for one to be better than the other, but for the sake of backwards compatibility I'd put it after the user's handler.

Reply:
> Currently it is basically the same as inside onPreviewKeyEvent after Compose. Like before this PR.

Not exactly. The code calling these methods is
```
onPreviewKeyEvent(composeEvent) ||
            scene.sendKeyEvent(composeEvent) ||
            onKeyEvent(composeEvent)
```
So before this PR, the backhandler would always be called (unless the user returned `true` from his own `onPreviewKeyEvent`). Now it is called only if the scene doesn't consume it too.

But as I mentioned above, I think we should preserve backwards compatibility and let the user's `onKeyEvent` first dibs on handling the event.

Reply:
there is no a big difference in the behavior, because in our case DesktopBackGestureDispatcher is in a root of the compose tree. But I think it is not fully correct to pass event down and if it is not handled then to call the navigation event

Reply:
> But I think it is not fully correct to pass event down and if it is not handled then to call the navigation event

Isn't the point of this PR to pass the event down and only run the back-handler if the event hasn't been handled?

The additional change Ivan and I are talking about is to also let the user-specified Window `onKeyEvent` handle it before the backhandler:
```
Window(
    onCloseRequest = { ... },
    onKeyEvent = { event ->
        if (event.type == KeyEventType.KeyDown && event.key == Key.Escape) {
            ...
            return@Window true
        }
        false
    }
)
```
With the current version of this PR, the logic above would be broken.

Reply:
the presented code must be broken because it behaves like: if **no-one** handles the esc-button event in **the whole app** then handle it.
the back handler, obviously, is a part of the app and it handles the event, so Window will receive nothing

Reply:
Ok, I've just now realized that the back handler itself is registered by calling `@Composable BackHandler()`, so it's not a global thing; has a specific location in the composition.

Now I'm no longer sure that it should be called from the window-level at all. Neither from `onPreviewKeyEvent` nor from `onKeyEvent`. Maybe it should be called from its place in the composition?

I'm also not sure about 
```
    protected val activeListener: BackGestureListener?
        get() = listeners.findLast { it.enabled }
```
Why is the last enabled listener used here? If I have two components with `BackHandler(enabled = true)`, why does the one that was added last to the composition win?

Reply:
> Why is the last enabled listener used here? If I have two components with BackHandler(enabled = true), why does the one that was added last to the composition win?

To match Android API that we do not control

Reply:
> To match Android API that we do not control

Sounds like wrong behavior/bug, if I'm understanding things correctly. If we remain with the behavior of listening to the event at the level of the window, I think it should be dispatched to the backhandler that became enabled last.

Reply:
After discussion we decided to go with `onKeyEvent(it) || backGestureDispatcher.onKeyEvent(it)`
cc @igordmn

Reply:
> After discussion we decided to go with onKeyEvent(it) || backGestureDispatcher.onKeyEvent(it)

Looks good to me